### ISSUE TITLE
Help, Questions and General Discussion

### ISSUE THREAD

**Issue by futpib** [futpib/policeman#18]
If you have a question, need help, unsure if something is an issue, have something else to say, whatever else — post it here.

**Comment by danieljl** [comment]
Is it possible to create a custom ruleset to block inline Javascript? Thanks.

**Comment by futpib** [comment]
@danieljl It is only possible to block "javascript:..." URLs, but not `<script>...</script>` tags.

**Comment by futpib** [comment]
From now on I'll build [pre-release versions](https://github.com/futpib/policeman/releases) before uploading anything to AMO. I know some people install them before they get reviewed and report issues, so now I encourage you to install those instead.

If it was not for cutting-edge-users and slow-ish AMO review process, people would get more broken updates, so thank you.

**Comment by bastik-tor** [comment]
@futpib Do you expect reports if the pre-releases work as expected?

E.g Are you going to assume that after x amount of time without a report of something broken it will be safe to release it to the general public? Or are you waiting until a fair share of people told you that the addon works as expected?

So far I don't notice anything broken. I tried to install two rulesets and it worked fine. Then I tested the HTTPS sandbox ruleset I installed. As far as I can tell it is working, as I don't see any images from non-https on https websites. Instead the addon displays placeholders. I hope this ruleset makes it into a future release.

**Comment by heforfree** [comment]
thank you for adding this test build
i try install rulset via "install link.." then it wont installed also probably same error on none prebuild
this link tested
https://gist.githubusercontent.com/somini/0a9e0406f03d5f14363a/raw/a7a4b939c97bc3a2406a3113e9be85c94cd75f86/magnet.ruleset

https://gist.github.com/heforfree/c39423c3eea947630eb0

and i tested allow_styles_same_site.ruleset and working very fine.Please add it as preinstalled ruleset
by default.
i just see this in ctrl+shift+j
syntax error allow_styles_same_site.ruleset:2

for https_sandbox.ruleset this need sample url

tested this also and working fine.
allow_my_very_special_script.ruleset

just if you can make an ruleset to reject per url path
http://site.com/ads/

**Comment by futpib** [comment]
@bastik-tor I think I'll just close issues I consider fixed, and if you see something is broken, open a new issue or comment on a mis-closed one. If no major issue pops up after some time I'll release it to general public.

**Comment by heforfree** [comment]
@futpib on 0.18pre1 no more syntax error and i am able to install ruleset via link

**Comment by futpib** [comment]
@heforfree Cool. Though that magnet-ruleset has no use as far as I can see, compatibility ruleset allows magnet links among other things.

**Comment by heforfree** [comment]
@futpib yes i remember bug fix for magnet link i selected that just for testing,since other ruleset is installed

**Comment by jyfo** [comment]
Under Popup, what does "show types if no request of this type was made"? What does checking that box do?

What is "content types available in popup"?

**Comment by heforfree** [comment]
@jyfo if it checked it is like below
![image 1](https://cloud.githubusercontent.com/assets/2247864/5604029/94b15154-93b8-11e4-951a-6aac03472dae.png)

if not checked 
![image 2](https://cloud.githubusercontent.com/assets/2247864/5604032/a8924660-93b8-11e4-9869-d207a8b54c68.png)

**Comment by futpib** [comment]
@jyfo Feel free to suggest a better title for that checkbox.

**Comment by jyfo** [comment]
Thanks heforfree and futpib, after staring at that screenshot for long time I finally reached enlightenment.

I think "Content types displayed in popup" might be better than "Content types available in popup" since checking or unchecking the boxes below it just effects what is displayed in the popup right?

To confirm, "show types if no request of this type was made" and "content types available in popup" just effects how the popup is displayed right?

**Comment by futpib** [comment]
@jyfo right

**Comment by esotericDisciple** [comment]
When I select all content types to be shown in the popup menu, the menu expands beyond my screen, making the "make temporary/make permanent" and other buttons inaccessible. Can you add a way for the menu to fit all display sizes/resolutions no matter how many items are shown in the menu, please?

**Comment by heforfree** [comment]
@esotericDisciple 
this is in to do list of developer  #40
In the meantime,you can uncheck "Show types if no request of this type was made".
this cause hide, zero (0) content type.so popup will be smaller

**Comment by Zylinder** [comment]
Just a word to remind that disabling Adblock Plus/Edge on the basis that Policeman handles external connections is a wrong approach. Many sites, more and more in fact, maybe because they realize how many of their external calls for ads/tracking are handled by blocking tools, have chosen to include those ads/trackers within their very own pages, with no external calls. In this scenario Policeman can do nothing. This is  why I believe Adblock Plus/Edge and Policeman are complementary and therefor both required.

**Comment by Corrodias** [comment]
I enabled "Replace with placeholder" under "Handle blocked objects". I am blocking all "objects". However, when I view a YouTube page, where the video would be is an empty, black rectangle. The menu correctly reports that the object is being blocked. I was expecting to see a light colored rectangle with the shield icon in it, as I have seen elsewhere.

**Comment by excalibur1234** [comment]
is it possible to use policeman INSTEAD of noscript and have the same (or even better) level of protection?
or does policeman complement noscript?

btw, thanks a lot for this great tool!

**Comment by Halibut80** [comment]
@excalibur1234 https://github.com/futpib/policeman/issues/93#issuecomment-67614507

**Comment by Zylinder** [comment]
Just installed Policeman 0.18 Public Release.

Great achievement. Those who have participated or at least followed the development my get things right, but for others as myself who discover new possibilities there may be a problem to understand how to achieve a new option.

I read "_Simple domains blacklist (just a file with domains) and hosts file format support_"
Problem for me, for any user of Policeman who did not attend the Policeman Github Winter Courses, is how to achieve this? I know how to add a Policeman formatted rule set, but how must one proceed to add a "_Simple domains blacklist (just a file with domains) and hosts file format support_"

Thanks

**Comment by heforfree** [comment]
@Zylinder you want add your url to Simple domains blacklist?
or you can find that  Simple domains blacklist file?

**Comment by futpib** [comment]
@Zylinder See https://github.com/futpib/policeman/wiki/Supported-file-formats

**Comment by heforfree** [comment]
@futpib but it not block them if it will origin?can you also provide way to block them completely?
and a one way to policeman wont show them on destination

**Comment by Zylinder** [comment]
@heforfree, @futpib, thanks but I've gone through what you mention already. What I do not know how to do is to **add** a simple domains blacklist or a hosts formatted file. 

If I go to Policeman Preferences, Import, I cannot import a text file, it's refused. It is this procedure link that I have not yet understood. I've spent a half/hour (nonsense compared to you guys' work), I'm not more retarted than the average (I hope) yet I miss something which must be obvious... except for me.

**Comment by heforfree** [comment]
just  in rulset manager select " install link"  you must select "raw"
https://raw.githubusercontent.com/futpib/policeman-rulesets/master/examples/simple_domains_blacklist.txt

**Comment by Zylinder** [comment]
@heforfree commented on [18 janv. 2015 16:05 UTC+1](https://github.com/futpib/policeman/issues/18#issuecomment-70411458 "2015-01-18T15:05:25Z - Replied by Github Comment Enhancer"):
> just  in rulset manager select " install link"  and add below link.
> 
> [https://raw.githubusercontent.com/futpib/policeman-rulesets/master/examples/simple_domains_blacklist.txt](https://raw.githubusercontent.com/futpib/policeman-rulesets/master/examples/simple_domains_blacklist.txt)
> 
> in below page you must select "raw" then add it as link or save it as ruleset and use install file
> 
> [https://github.com/futpib/policeman-rulesets/blob/master/examples/simple_domains_blacklist.txt](https://github.com/futpib/policeman-rulesets/blob/master/examples/simple_domains_blacklist.txt)

Thanks but :  
1- I find no "Install link"  
2- What do you call the Ruleset Manager?
3- I wish to add either a home-made domains blacklist, how to add it?

Forget it, this is beyond me.

**Comment by heforfree** [comment]
1+2 i mean Manage Rule set.where the ruleset are.and on bottom of that you can find  install link
about:policeman#rulesets-manager
3.you can save the simple_domains_blacklist.txt
and give it any name just and i dont see  any http or www so just add other part of url

**Comment by futpib** [comment]
@Zylinder 
![](https://cloud.githubusercontent.com/assets/4330357/5792900/09485670-9f3f-11e4-962e-8adfcabc7905.png)

I guess our UI is too fancy.

**Comment by bastik-tor** [comment]
@Zylinder for the blacklist I was able to create ```some_filename.txt``` with ```example.com``` ```another-example.org``` as content, one domain per line and install it. For me it was not rejected.

The hosts file should work similar, though I have no interest in including my hosts file in policeman.

**Comment by Zylinder** [comment]
@heforfree @futpib @bastik-tor , great news : I finally made it. Mama mia. A thousand and one thanks to all of you, really nice to help the blind man I was...

1. I was reasoning on the basis of an import/export rules, so I proceeded accordingly from Options/Import
2. I had never noticed the Install file - Install link buttons in the Manage Rule Sets area.

Wow. Coffee? Thanks a lot. This is not the first time I miss something which is right under my eyes. Nevertheless I had a lack on the Options construction. So : I feel better, I hate it when I feel more retarted than the average (if Average is in  the room, it wasn't meant for you!)

**Comment by ozar** [comment]
Hello, I'm having an issue here with Policeman .18

In General Preferences:
If I set "left button click" to Open popup, the window opens but it's very small, cutting off some of the domain information.  Sometimes, clicking an one of the drop down arrow buttons makes the window go ahead and open fully revealing the hidden info.

If I set "On hover" to Open popup, the window opens fully, showing all domain info, as it did in the previous Policeman release.

Am I doing something wrong, or is this the expected behavior of the latest release?  Sorry is this is covered elsewhere and I've missed it.  Thank you.

**Comment by Zylinder** [comment]
Hum... I've installed a simple test 2 item blacklist file but neither one of the 2 urls are blocked.

Options/Manage rule sets/Installed sets : Blacklist of 2 domains (Enabled) - OK
and a xyz.ruleset in my Firefox profile \policeman\rulesets - OK
xyz.ruleset is stricly identical to my source file (test.txt)

File is composed of : 
# Test
neowin.net
melty.fr

But neither neowin.net (called from bookmarks) nor melty.fr (called from a site) is blocked...
Any idea or is it -- if not me -- my Firefox which is problematic?

**Comment by heforfree** [comment]
@Zylinder  as #63 still open so it should not work  and for me to not working

**Comment by Zylinder** [comment]
OK @heforfree thanks -- We'll remain on standby then.

**Comment by futpib** [comment]
@Zylinder @heforfree Well, I thought they were working OK, otherwise I wouldn't mention them in release notes.

**Comment by futpib** [comment]
@ozar Actually I've seen that with Nightly. It ran on Xephyr and I mistakenly thought it was his fault somehow, but now that you mention it, I reproduced it on my Xorg. What Firefox version are you running?

Opened #157.

**Comment by Zylinder** [comment]
@futpib commented on [18 janv. 2015 20:56 UTC+1](https://github.com/futpib/policeman/issues/18#issuecomment-70422900 "2015-01-18T19:56:03Z - Replied by Github Comment Enhancer"):
> [@Zylinder](https://github.com/Zylinder) [@heforfree](https://github.com/heforfree) Well, I thought they were working OK, otherwise I wouldn't mention them in the release notes.

Being able to use rulesets built on the source of a simple text file is very nice but personally I consider it as a true extra feature far ahead of RequestPolicy's concept. I mean, IMO, this extra feature is both a lot of work and beyond the scope, the concept of filtering what websites communicate one another. But, you're doing it and it'll come out fine. Tougher is the aim more the incidents arise before the total reliability, we all know that. But looks like the toughest is accomplished, the stage now is that of polishing, as I see it.

Policeman is really a very nice work, not only does it continue where predecessors seem to have stopped but moreover it ads extra features, many. I'd just love to help more, should it be by sharing more testings. Point is I don't encounter issues or when I do i come here and find out they're already being processed if not resolved with regular updates betas included.

So : you have a bunch of guys and gals (I hope!) thankful for your work, your commitment to a better Web.

This is the end :) ... of the chapter!

**Comment by futpib** [comment]
I'm thinking to allow all the requests to Firefox's built-in schemes (what's called "chrome destination" in popup) by default, since there is really no networking involved at all. It seems, this requests are really of no concern for a user, but having them blocked may break stuff (#142 and also some thing in compatibility ruleset). Does this make sense? I mean, will anybody miss "chrome destination" and if you will, then why?

**Comment by somini** [comment]
It would certainly simplify things.

**Comment by ozar** [comment]
**@futpib**
Yeah, my Firefox is 35.0 on Linux.  I see you've opened a bug report regarding this issue and closed it already with a commit.  Thanks for checking it out and fixing it.  I really do like this extension and certainly appreciate your hard work on it.

**Comment by heforfree** [comment]
@futpib as you no send or recive happen allow them if you want.

**Comment by Zylinder** [comment]
Here I've allowed Everything-All to Chrome Destination when I discovered it solved an issue I had encountered with RequestPolicy, that of RSS feeds. Not allowing Chrome Destination would block the RSS available feeds on a website to show (to popup from the RSS button). There must be other advantages but this is the only one I encountered.

Concerning Chrome Origin -> websites (all?) I haven't allowed it yet, even if I've encountered once and once only an issue with an add-on (Blockulicious) when sending the LicenseID for an upgrade : the connection to their server required Chrome Origin -> All (couldn't test with their server only as the connection could not appear in Policeman's popup since it did not open a page).

What are the practical consequences in terms of privacy be it Chrome destination or Origin is hard to understand for a non techie. I can say that allowing Chrome Destination shows on many sites that doing so lets the site open several about:blank pages... and so what ?

At this point we are dealing with deeper concerns, at least for the average user.

**Comment by requiredregistration** [comment]
i use [Chrome destination] to stop automatic playback in youtube

**Comment by Zylinder** [comment]
@requiredregistration commented on [19 janv. 2015 12:02 UTC+1](https://github.com/futpib/policeman/issues/18#issuecomment-70476731 "2015-01-19T11:02:12Z - Replied by Github Comment Enhancer"):
> i use [Chrome destination] to stop automatic playback in youtube
> 
> edit: i allow it temporarily to start the video playback manually

Now that's interesting. I'll have to give it a try, especially now that Youtube has changed recently its playback management.

**Comment by heforfree** [comment]
@futpib  i must ask you to sure,which request are blocked to chrome destination?
i again go to youtube  and set reject youtube.com to chrome destination and but i see
it allow even with reject.so i think it allready done

**Comment by somini** [comment]
Just change this in the `about:config`
   media.autoplay.enabled

**Comment by atomGit** [comment]
wow - thanks for your work! finally, a very attractive alternative to the largely defunct Request Policy :)

i am really liking this extension - i like the clean and simple, yet powerful pop-up UI

couple small things i noticed...
the pop-up UI gets cropped when it fills the height of the screen, but not the width, and a vertical scroll-bar is drawn (the right edge of the UI is covered with the scroll-bar)

! would be nice to override a global rule (rule set) on a per-site basis - example: i don't want to be annoyed by having to allow site-to-site requests for every site, yet on a few i might want to disallow JS (same domain to same domain)

consider allowing a R-click on tool-bar icon with the same options as for other click events

consider allowing to edit the persistent rules - maybe dbbl click displays the rule in the already available combo/input controls at the top so it can be edited and saved - i'm also wondering if allowing RegExp's in the URI's might be useful

documentation is needed - i'll mail you about that (i can help)

i can't believe how well this sucker looks and works and it's only at v0.1* - very well thought out

THANKS again!

**Comment by aimtsou** [comment]
Two questions for the developer: 
1) Why some scripts appear to be called multiple times instead of one time?
2) Is it possible to disable one script instead of all? Will it be available in a later version?

**Comment by Halibut80** [comment]
@aimtsou 
1)
Maybe this <a href="https://github.com/futpib/policeman/issues/154">https://github.com/futpib/policeman/issues/154</a>
2)
<a href="https://github.com/futpib/policeman/issues/81">https://github.com/futpib/policeman/issues/81</a>
For now you can make <a href="https://github.com/futpib/policeman-rulesets/blob/master/examples/allow_my_very_special_script.ruleset">custom ruleset</a> for that.

**Comment by esotericDisciple** [comment]
I'm now on Xubuntu 14.10 with the node packaged (npm) version of coffee-script.  The build script is not working at all. Judging by the way the text looks when I open it in "mousepad" with a "kate" color scheme, it appears to be something about the syntax that isn't compatible with my system. How can I fix that?

**Comment by futpib** [comment]
@esotericDisciple What exactly does it say when you execute it?

**Comment by esotericDisciple** [comment]
If I run 'bash build.sh --pack' it just shows the function printouts. If I run 'sh build.sh --pack' I get the following two messages:
build.sh: 46: build.sh: Bad substitution
build.sh: 47: build.sh: Bad substitution

**Comment by futpib** [comment]
@esotericDisciple It's a bash script, so what is the output of `bash build.sh --pack`?

**Comment by esotericDisciple** [comment]
"Performing clean...
Compiling coffee to js..."
Nothing else is said...or done, it seems.

**Comment by futpib** [comment]
@esotericDisciple Can `coffee` be found in your PATH?

**Comment by futpib** [comment]
@esotericDisciple Well, I have no idea what is going wrong. Can you please run `bash -x ./build.sh` and send the output you get to me?

**Comment by esotericDisciple** [comment]
````
bash -x ./build.sh -p
+ source_root=src
+ build_root=build
+ do_build=true
+ do_pack=false
+ do_svg=true
+ main -p
++ getopt -o hcips:b: -l help,clean-only,keep-icons,pack,source-dir:,build-dir: -- -p
+ args=' -p --'
+ eval set -- ' -p --'
++ set -- -p --
+ '[' '!' -z -p ']'
+ case "$1" in
+ do_pack=true
+ shift
+ '[' '!' -z -- ']'
+ case "$1" in
+ shift
+ break
+ build
+ clean
+ echo 'Performing clean...'
Performing clean...
+ '[' true = false ']'
+ rm -rf build
+ '[' true = true ']'
+ generate
+ copy
+ rsync -qav '--exclude=.*' src/bootstrap.coffee src/chrome src/chrome.manifest src/defaults src/icon.svg src/install.rdf src/lib build
+ make_coffee
+ echo 'Compiling coffee to js...'
Compiling coffee to js...
++ find build -type f -name '*.coffee'
+ coffee_files='build/lib/content-policy.coffee
build/lib/ui/content-context-menu.coffee
build/lib/ui/popup.coffee
build/lib/ui/about-policeman.coffee
build/lib/ui/panelview.coffee
build/lib/ui/load-overlay.coffee
build/lib/ui/ui.coffee
build/lib/ui/popup-tk2.coffee
build/lib/ui/toolbarbutton.coffee
build/lib/l10n.coffee
build/lib/prefs.coffee
build/lib/updating.coffee
build/lib/ruleset/errors.coffee
build/lib/ruleset/manager.coffee
build/lib/ruleset/domains-blacklist-ruleset.coffee
build/lib/ruleset/persistent.coffee
build/lib/ruleset/tree-nodes.coffee
build/lib/ruleset/hosts-ruleset.coffee
build/lib/ruleset/temporary.coffee
build/lib/ruleset/in-memory-ruleset.coffee
build/lib/ruleset/base.coffee
build/lib/ruleset/tree-ruleset.coffee
build/lib/ruleset/format-registry.coffee
build/lib/color.coffee
build/lib/windows.coffee
build/lib/blocked-elements.coffee
build/lib/mutable-ordered-map.coffee
build/lib/utils.coffee
build/lib/request-info.coffee
build/lib/file.coffee
build/lib/request-memo.coffee
build/lib/require.coffee
build/lib/tabs.coffee
build/chrome/content/xul-page-common.coffee
build/chrome/content/preferences/rulesets-manager.coffee
build/chrome/content/preferences/popup.coffee
build/chrome/content/preferences/user-rulesets-import.coffee
build/chrome/content/preferences/user-rulesets-import-requestpolicy.coffee
build/chrome/content/preferences/user-rulesets.coffee
build/chrome/content/preferences/user-rulesets-toolkit.coffee
build/chrome/content/preferences/index.coffee
build/chrome/content/preferences/general.coffee
build/chrome/content/preferences/user-rulesets-export.coffee
build/bootstrap.coffee'
+ for f in '$coffee_files'
+ coffee -cbp build/lib/content-policy.coffee
+ tee build/lib/content-policy.js
+ coffee_status=1
+ '[' 0 -ne 1 ']'
+ return 1
+ return 1
+ return 1
+ return 1
+ exit 1
````

**Comment by futpib** [comment]
@esotericDisciple Very strange, looks like `coffee -cbp build/lib/content-policy.coffee` fails, but prints nothing to stderr. I have npm coffee-script-1.8.0 installation too, and it works fine for me. Next step would be to find out why exactly `coffee` fails. I believe I can't help you with that, sorry.

**Comment by esotericDisciple** [comment]
I finally found out what the problem was; it was related to the same issue as this one:
https://stackoverflow.com/questions/13461638/installing-coffee-script-on-ubuntu-installs-all-the-files-correctly-but-command
Uninstalling the Ubuntu package "node" and manually linking /usr/bin/nodejs to /usr/bin/node allows the build.sh script to run successfully.

**Comment by Ulf3000** [comment]
Hi i registered because i wanted to adress the situation with other adblockers.

" I thought about supporting easylist-format (or whatever it's called, I mean the rules format adblock uses) from the start, if that's what you ask about. I think there is no real benefit in supporting it, since we already have adblock who does it well, and policeman can coexist with adblock. 

Honestly, unless you have a good argument, I won't implement it anytime soon."

i made a new profile and tried every adblocker on Mozilla page and the problem is sometimes the adblocker runs before your addon and sometimes it doesn´t. best would be if it runs after the adblocker because then it doesn`t show  all the pesky doubleclick goglesyndication entrys and stuff the adblocker should already be done with, which obfuscates policemans widget and makes it hard to read/controll.

a real downside and probable bug is:
adblockers use rules to re-format the page and how to hide objects to make it look nice , if policeman runs before the adblocker(90% of times) , these rules can´t apply since policeman already caught the adservers first and the adblocker does nothing in this case(not all the time but happend to me quite frequently)and the site has wholeswith no content or content cramped together!
so thats a real incompatibility and major pain imho.

so either it would be enough to force policeman to run after the adblocker(then all ad and tracker won´t show up in policeman and style rules apply corretly). 
that's actually the best solution, everybody can choose his favourite adclocker. but i read its not really possible to change order of load of addons. 

so if thats not possible you could really implement the easylist and then your addon would all the more enjoyable imho. blend tracker entrys out or recolor them to violett or something like that , would be badass imho and really helpfull!

already thanx for this piece of beauty though , like i said on mozilla reviews i wanna go with policeman alone , my techy friends were all on fire too when i showed them your addon.

its really better than the other addons. 
for example i´ve chosen img and css for  anyone to anyone. so i can use forums and stuff correctly(porn and ads then get stil caught by the adblcoker and make the internet at least readable).  
on my anime streaming site i´ve only allowed one video hoster so the pages load faster. 
on another site i allowed all hosters though , since they are often down. this is badass and the functionality of noscript  nor other addons can do that so conveniently with filtering for media images scripits , its actually much faster to configure for correct browsing, a new noscript installation takes like weeks with endless allowing and blcoking and reloading lol^^! this will be may main security addon for the foreseeable future , i checked on inline scripts but even the worsts cracking sites and even fake warez sites don´t have any problematic or any at all inline scripts as far as i could see. just onclicks and other easy stuff so that is not of my concern. firefox is sandboxed.

i love you man lol XD

ps : scrolling for the widget should be implemented so its doesn´t grow over a specific size when watching the single entrys in the filter view

**Comment by somini** [comment]
µBlock has a version for Firefox now, there no need to use bloated AdBlock Plus anymore.
https://github.com/gorhill/uBlock/releases
I can confirm it coexists peacefully with Policeman.

**Comment by Ulf3000** [comment]
A temporary fix for the problem i described above is to , disable policeman before closing firefox and next time it loads up , enabling  policeman. since i rarely ever close firefox thats OK for me right now , still i would like to know a way to delay the startup of policeman and everything would be good  i even thought about implementing a file access delay on the xpi, but found nothing to achieve that.

**Comment by futpib** [comment]
I believe that #54 is about the same problem @Ulf3000 is talking about. (Just mentioning it here to make these notes easier to find)

**Comment by Ulf3000** [comment]
nope thats not my problem, 
my problem is simply that i don´t want to block ads with policeman... so policeman has to load after the adblocker addon loads, 
a temporary fix is to start firefox with policeman disabled and then enabling it afterwards!

**Comment by Zylinder** [comment]
@somini commented on [30 janv. 2015 12:08 UTC+1](https://github.com/futpib/policeman/issues/18#issuecomment-72185985 "2015-01-30T11:08:17Z - Replied by Github Comment Enhancer"):
> µBlock has a version for Firefox now, there no need to use bloated AdBlock Plus anymore.
> 
> [https://github.com/gorhill/uBlock/releases](https://github.com/gorhill/uBlock/releases)
> 
> I can confirm it coexists peacefully with Policeman.

Quite true. I've installed it for testing (after having backuped my FF profile and removed AdBlock Edge) and it runs very well with Policeman. uBlock & Policeman is a winning team. I'm keeping uBlock because of obvious advantages over Adblock, in the same way Policeman has over RequestPolicyContinued.

In my opinion it is not Policeman's role to block ads, its function as I see it is to manage external calls a site performs when called. I understand that futpib has gone on a lot of extra work to satisfy requests for extra filtering, but this extra is the work of a blocker, Adblock or uBlock. 

So, with Policeman handling the esternal calls and uBlock managing what is loaded, where is the problem? I can say also that with Policeman in the arena uBlock has much, much work to perform when indeed remains under its power what is from the site itself or what is included with what the user will have authorized to be called.

Let's not make things complicated. An add-on, an application, a software starts with a simple idea, then it gets bloated not if it improves itself but when it skips on the temptation to add functions (with sometimes the idea to access a "universal" range of possibilities) that often, so often lead to a bloated application and add confusion to most users who wish a good product aimed at what it is intended for.

**Comment by Ulf3000** [comment]
@zylinder... the problem is that your adblocker won´t do shit because you already didn´t allow any connections to adservers in policeman anyway, you can easily check that with an adblcoker which shows the number of blocked items ... instead these adservers show up the ui of policeman with unneccessary entrys and you don´t even need adblocker anymore...

**Comment by Zylinder** [comment]
@Ulf3000 ...  not all ads/trackers or whatever is considered as intrusive come from external sources, some sites, more and more in fact include all this nasty extra data in their own pages, or in content providers with the same host (allowing goodsite.com allows craps.goodsite.com, beware of that if you've allowed default Policeman rule "Same Site").

The idea is : Policeman will block what is called externally ; we all know that - Blockers will block what's landed on the page after Policeman and what didn't have to land because it was there in the first place.

It's as simple as that, and this is why I state that it is not Policeman's role to be an adblocker (more than what it implicitely performs), simply because not all intruders are foreigners (to the site!). The problematic is at the intersection of domestic and foreign action : Policeman would be the CIA and an adblocker the FBI ... :)

**Comment by somini** [comment]
I agree with @Zylinder here. Not that uBlock exists and works well in tandem, putting ad blocking inside Policeman is just needless work for little gain.
Not to mention ad blocking also involves CSS manipulation, which IMHO goes beyond the scope of what Policeman is trying to do.

**Comment by Ulf3000** [comment]
@zylinder , i use "http user agent cleaner" addon to surf only in html/css mode and to restrict cookies and plugins/mimetypes , and enable stuff only for specific sites.

i also disagree, i think policeman and an adblocker are pretty much the same thing in that they both block specific rule-centric content from 3rdparty providers and do nothing else. but i want one to handle the ad and trackers and the other to handlethe leftover real cross-content

**Comment by Ulf3000** [comment]
@somini 
yes thats it, and thats why the adblocker has to start before policeman because otherwise all ads will be blocked by policeman and not a single one will even reach your adblocker. thats what im talking about, i´m not wishing for policeman too include adblcock it AT ALL,only IF theres no other way to solve the problem. 

this is the same site loaded with adblocker before policeman , everything is nice and clean and policeman only shows the videohosters becasue ads and tracker were already removed by the adblocker
http://i.imgur.com/prpKLR1.jpg
![untitled-1](https://cloud.githubusercontent.com/assets/10765339/5989059/49f3c910-a972-11e4-950c-bba1408aaa72.jpg)


this shows adblocker loaded after policeman! policeman is now obfuscated with annoying rubicon adthis and adcash and google analytics, so they won´t even touch your adblocker. this is sadly the default behaviour 
http://i.imgur.com/BraKK1d.jpg
![untitled-2](https://cloud.githubusercontent.com/assets/10765339/5989088/5c7bef1c-a973-11e4-9a1c-7a0334854c62.jpg)

**Comment by somini** [comment]
@Ulf3000 Uh. Now I see what you mean.
Is there a way to control the addon load order? How did you got those screenshots?

**Comment by Ulf3000** [comment]
the load order is to restart firefox WITH DISABLED policeman and when firefox has loaded up all addons and ui you go to addons and enable policeman. then you´ll get the first screenshot.

restarting with enabled policeman gives you the second screenshot.

the load order is important for the restart-less addons from what i´ve experienced.
i think the first screenshot should be the default behavior.

**Comment by futpib** [comment]
@Ulf3000 But how is this different from #54?

**Comment by futpib** [comment]
Anyway, opened #170 for this, please go there for any farther discussion.

**Comment by k-h** [comment]
I often have odd fonts not showing.  On this page Policeman says no rejected requests but many buttons look like this:https://i.imgur.com/y4kII8s.png .  Request policy used to let me allow fonts and other objects.  Can Policeman do this?

**Comment by futpib** [comment]
@k-h Works for me ![_7](https://cloud.githubusercontent.com/assets/4330357/6037572/30b48ee8-ac66-11e4-8184-cd04f4ae5a17.png)

Check "Font" checkbox on `about:policeman#popup` page or look for the font requests under the "Other" button.

**Comment by k-h** [comment]
I just realised I have Random Agent Spoofer set to "use standard font set" so it's probably my fault.

**Comment by b10018** [comment]
Dear futpib.
I used to use request policy. I use it now. This add-on is incredible! I'm released from the troublesome work of selecting the updating software. Thank you.
I'd appreciate it if you could answer the following questions. 
policeman-rulesets/general/
・allow_styles_and_images_same_site.ruleset
・allow_styles_same_site.ruleset
Do these work normally?Does the watch work normally?

sample(tested http://www.sony.net/)
![sss](https://cloud.githubusercontent.com/assets/10888882/6088946/3812fbda-ae9f-11e4-97c8-92a7a0d36fc0.jpg)
![ss](https://cloud.githubusercontent.com/assets/10888882/6088948/391fd516-ae9f-11e4-87ed-adbdf00a1a94.jpg)

Regards,
b10018

**Comment by futpib** [comment]
@b10018 Yea, they both seem broken. Try older revisions ([allow_styles_and_images_same_site](https://github.com/futpib/policeman-rulesets/raw/a97e53b013c3c585d5ed304ece721b89a2a9142d/general/allow_styles_and_images_same_site.ruleset) and [allow_styles_same_site](https://github.com/futpib/policeman-rulesets/raw/0221316e3869b34e93a22c4c5348f555fee06470/general/allow_styles_same_site.ruleset)). Downside is that they are not TLD-aware (co.uk and such).

**Comment by anonish** [comment]
futpib: thanks for this great addon, as others stated it works great in tandem with ublock
my request: please provide some kind of scrolling mechanism on the popup.. on my laptop in portrait/tablet mode, screen width goes to around 800px, and the addon (if the domain is long enough) goes off-screen and is impossible to enable/disable rules as the button disappears.  (sadly this happens with many add-ons).  is there a way to implement a horizontal scrollbar?

**Comment by Halibut80** [comment]
@anonish To userChrome.css or Stylish:
<code>@namespace url(http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul);
#policeman-popup > arrowscrollbox > stack > hbox  {
  max-width: 770px!important;
  overflow-x: scroll!important;
}
#policeman-popup  {
  max-width: 780px!important;
  max-height: 550px!important;
}</code>

**Comment by anonish** [comment]
@Halibut80 
many thanks my good sir, that does the job nicely

**Comment by linuzzz** [comment]
Hello, first my thanks for the great job!
I would like to know if it is possible to change the default "same site" and "same second level domain" rules in order to just allow image and styles instead of anything. Thanks!

**Comment by Halibut80** [comment]
@linuzzz Just disable them, and install ruleset from this <a href="https://github.com/futpib/policeman/issues/18#issuecomment-73332701">post</a> ("install link" in about:policeman#rulesets-manager page).

**Comment by futpib** [comment]
@b10018 I said they are broken, but now as I looked into it I see clearly they work fine. It's aether a very nasty bug, or I was mistaken (I actually think I just forgot to hit "Save" that first time).

**Comment by RyanTG** [comment]
Great addon, thanks. It speeds up page-loading times considerably.
My question: When I'm logged into gmail, and then open up google maps in a new tab, it shows me logged in on the new tab. I would love it if that didn't happen. But it's unclear what to reject to accomplish that. Do you know? Perhaps this is a better task for disconnect or ghostery, since it's likely cookie-related.

**Comment by heforfree** [comment]
@RyanTG for now use Priv8 addon
https://addons.mozilla.org/en-US/firefox/addon/priv8/

**Comment by RyanTG** [comment]
@heforfree Thanks - looks great!

**Comment by RyanTG** [comment]
Last question: Can you give advice on whether it's redundant to use Policeman AND a tracker-blocking addon like Ghostery, Disconnect, or Privacy Badger? It seems that since Policeman can block all third parties, then the trackers/cookies can't access you. But maybe I'm missing something. Maybe those other addons can function as back-ups (memory-intensive back-ups!) in case I accidentally accept a third party file that contains a tracker.

Thanks.

**Comment by heforfree** [comment]
@RyanTG  you can keep them for when accidentally accept a third party request,but it almost trail and error and when issue  #166 completed it make even more easy

**Comment by anonish** [comment]
what's the css for links (ie the allowed/rejected clickable items, currently blue)?

**Comment by Halibut80** [comment]
@anonish <code>#policeman-popup .text-link</code>

**Comment by anonish** [comment]
@Halibut80 bah gawd where has that been all my life! thanks!

**Comment by vdcbb** [comment]
futpib, this is a fantastic extension. Very nice work!

But I keep wishing there was a keyboard shortcut to open Policeman. Like I could tap CTRL+SHIFT+Z or something and the popup would open and then I could use the mouse within the popup to make my selections. As is, I have to travel the length of the screen with the mouse, activate the popup, then travel into the popup. Lot of mouse movement is involved--and the process is slow.

Thought I'd offer this request in the hope you are receptive. The keyboard shortcut could be built-in (e.g., NoScript opens with CTRL+SHIFT+S by default) or maybe an option could exist to activate a shortcut or something in Preferences > General Preferences > Toolbar button behavior?

**Comment by heforfree** [comment]
@vdcbb #85

**Comment by ozar** [comment]
Guys, I've installed 0.19pre1, but all version indicators within the extension are showing that 0.18.1 is still installed, such as in about:addons, policeman pop-up, preferences page, etc.

Can any of you relieve some confusion on my end by confirming that this is the intended behavior of pre-release versions?  If not, perhaps I did something wrong when installing this version.

Thank you.

**Comment by Halibut80** [comment]
<blockquote>I've installed 0.19pre1, but all version indicators within the extension are showing that 0.18.1 is still installed</blockquote>
Same.

**Comment by ozar** [comment]
@Halibut80

Okay, thank you for checking...

**Comment by futpib** [comment]
@ozar I forgot to change the version string. Replaced the package with a fixed one now.

**Comment by ozar** [comment]
@futpib
Now all is good... thank you!  :)

**Comment by b10018** [comment]
Hi, I I've installed 0.19pre1.
I don't know how to stop the redirection warning. I want to stop it.

**Comment by AtTheGates** [comment]
Hello,

I use the "reject_special_paths" ruleset with the method "Block any path that contain "/ads/" or "/adv/".

Would it be possible to block these entries of a whole URL? So not just in the path? I had several tries, but don't get it. :(

Thank you in advance.

**Comment by heforfree** [comment]
@AtTheGates 
http://example.com/images/2015/2/ads/15-2-26-1737415.jpg
this request have ads in path
then Policeman will reject  this request not just for after ads/ path
if request #81 completed then it make an easy to block

**Comment by AtTheGates** [comment]
Yes, I know how to block entries in path. I use a ruleset with ~60 entries. For example:

rules:
  web -> web:
    [path]
    -> * "/ad." *: REJECT
    -> * "/ads." *: REJECT
    -> * "/ads/" *: REJECT
    -> * "/ads_" *: REJECT
    -> * "/adtags." *: REJECT

My question was if I could use these entries to take affect in a whole URL. Not just in the path. Something like "prePath" AND "path".

If I have the entry "    -> * "pixel." *: REJECT" in my ruleset the following URL should be blocked:

http://pixel.example.com/images/2015/

**Comment by futpib** [comment]
@AtTheGates `[spec]` might be what you are looking for, it will match against the whole URI. Policeman (mostly) follows Mozilla's [nsIURI](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIURI) conventions for naming different components of URIs:
![Components of a URI](https://developer.mozilla.org/@api/deki/files/3267/=uri-diagram.png)

Unfortunately `spec` is not shown on the (otherwise very descriptive) picture above. `spec` is the whole URI.

**Comment by AtTheGates** [comment]
Ah, wonderful! Thank you very much, **futpib**. I wasn't aware of the [spec] parameter. It works perfectly now. :)

Greets

**Comment by w0lfe** [comment]
Something about youtube seems to have changed, before I just had to the xyz.googlevideos.com cdn for a video and my other defaults work, but now even if I manually allow everything the video will not load. I have to suspend  on this tab to get the video to load. Anyone else notice this change?

**Comment by CrisBRM** [comment]
Still no manpages? I'd like documentation regarding the rule syntax, in order to write my own rules, namely, one that accepts all 1p --> 1p requests (save for scripts, and pings), whilst rejecting (at least by default) 1p --> 3p requests.

**Comment by futpib** [comment]
@CrisBRM [On "Other" and "Uncat."](https://github.com/futpib/policeman/issues/101#issuecomment-69667754)

**Comment by OdinGitDat** [comment]
Ive tried everything i can to make policeman work with the http://alientube.co/ addon for firefox. Suspend on tab does not work. The only thing that seems to work is suspend policeman entirely. Do i need to create a custom ruleset for addons accessing websites?

**Comment by futpib** [comment]
@OdinGitDat Works for me with the [latest pre-release](https://github.com/futpib/policeman/releases/download/0.19pre1/Policeman_0.19pre1.xpi).

**Comment by OdinGitDat** [comment]
oh right it works with v.19 (i was on .18), thank you!

**Comment by k-h** [comment]
Hi,

I'm trying to get http://earth.nullschool.net to work. In policeman I let the site do everything but I get the message:

canvas.getContext(...) is undefined
and then
context is undefined

Tried this on a MacOS running 37.0.2 and Windows 7 with firefox 37.0.1

k-h

**Comment by coffeeface** [comment]
How would I block individual files from a domain that I otherwise trust? For example, for one particular site, I would like to allow ALL scripts to run except for "example.com/js/ads.js" and "example.com/js/ad.js"

**Comment by Halibut80** [comment]
@coffeeface <a href="https://github.com/futpib/policeman-rulesets/blob/master/examples/allow_my_very_special_script.ruleset">https://github.com/futpib/policeman-rulesets/blob/master/examples/allow_my_very_special_script.ruleset</a>

**Comment by coffeeface** [comment]
Thank you - I've created a rule based on that ruleset, but it fails to install:

magic: policeman_ruleset
version: 0.1
id: "forbid_specific_script"

l10n:
    en-US:
        name: "Forbids specified scripts"
        description: ""

rules:
    [spec] * -> "https://example.com/js/ads.js": REJECT

**Comment by Halibut80** [comment]
@coffeeface try:
```

magic: policeman_ruleset
version: 0.1
id: "reject_special_script"

l10n:
  en-US:
    name: "Blocks some scripts"
    description: ""

rules:
  web -> web:
    [spec]
    * -> "https://example.com/js/ads.js": REJECT
```
work for me.

**Comment by maeludir** [comment]
Hi I love the addon, and I've been using it reliably for quite a while.

But I've got a fresh install here, and while I've imported my old profile okay (which included all my rules) Policeman doesn't seem to be working - clicking on the shield icon shows no requests at all, as if I'm not on a website.

**Comment by vitaminx** [comment]
Hi, I'd love to synchronize my Policeman rules between several computers - I use unison (like rsync), but I can't find the files where the Policeman rules are stored.
There's a directory called "~/.mozilla/firefox/default/policeman/rulesets" but it's empty.
Which files do I need to synchronize? Where does Policeman store it's rules?
Thanks.

**Comment by vitaminx** [comment]
Hi, I'd love to synchronize my Policeman rules between several computers - I use unison (like rsync), but I can't find the files where the Policeman rules are stored.
There's a directory called "~/.mozilla/firefox/default/policeman/rulesets" but it's empty.
Which files do I need to synchronize? Where does Policeman store it's rules?
Thanks.

**Comment by futpib** [comment]
@vitaminx `policeman/rulesets` is for installed rulesets, other prefs are in `prefs.js` (under `extensions.policeman.*`)

**Comment by WagnerGMD** [comment]
I believe, there's a little mistake with the latest release (0.19pre3). It's simply a syntax error or the wrong file ?
Yesterday, I download the addon twice. And on the page about:addons, Policeman didn't report the right version. I saw the previous (0.19pre2).
Also I checked the file **install.rdf** and I saw this ligne (n°7) : `<em:version>0.19pre2</em:version>`.

**Comment by pberenyi** [comment]
Hi,

HTML5 playback on Youtube is not working with Policeman if there is no rule to allow youtube.com -> any domain. If I load any Youtube video, and set the rules to allow every listed domain, and Policeman icon becomes full green and no rejections are shown, even then the video is not loading. Once I add the above mentioned any domain rule, the video loads properly. Any ideas how to resolve without allowing everything?

Thanks

**Comment by Halibut80** [comment]
@pberenyi Known bug: <a href='https://github.com/futpib/policeman/issues/196'>incompatibility with YouTube HTML5 DASH playback</a>

**Comment by ImN1** [comment]
Great addon!
When sites that with a long long domain name, the [add rule] button go outside the screen...
UI must be fixed with line wrapping or a scrollbar.

**Comment by acrobat1** [comment]
Does the extension use absolute paths to find its rules location ? 
After moving my Firefox profile I couldn't find or install any ruleset, after resetting 
**extensions.policeman.manager.installedPathsByIds** it started working again.

**Comment by futpib** [comment]
@acrobat1 Shame to admit, but the latest release (0.18.1) does. This was fixed in a2431aa445885d8c87e3ce4b337dc5d8ae548cc5

**Comment by acrobat1** [comment]
OK thanks, I  had to scratch my head a bit to understand why all rules were lost  but I figured it out in the end...
Great extension, BTW.

**Comment by LucaLawZ** [comment]
Out of interest why are some networks that are used for privacy blocked as default in the standard ruleset?

Surely for those that use them networks wouldn't FP be an enhancement?

**Comment by heforfree** [comment]
@futpib
is there any news for update addon,at least for defect and bug

**Comment by dxdragon** [comment]
Firstly, Thanks futpib to bring us so great addon!! However, Sometimes I can't get help due to no documention. Now I'm trapped.
Problem: When I visit some online video websites, e.g. www.youku.com, they request some objects from IP sites with random digits, my current solution is
Allow (objects) youku.com (or other domains) —> [Any domain]
as you know, some unsafe objects will skip.... So I have to create a ruleset.

rules:
  [contentType] OBJECT:
    [schemeType] web -> web:
      [baseDomain] "youku.com" -> /\d+\.\d+\.\d+\.\d+/: ACCEPT

But It doesn't work! So how to write the relevant rule?
Thanks very much

**Comment by bizaff** [comment]
After not restarting firefox for a while, sometimes the popup becomes a fixed size.  When I have longer domains, the popup is too small and doesn't show the "Add rule" buttons.

**Comment by acrobat1** [comment]
I've noticed that the **policeman** folder inside my Firefox profile is actually empty, rulesets (as far as I can see) are not stored there but in the prefs.js file.
Is that intended behaviour?

If  so, it would be nice IMO to also store permanent rules somewhere in the **policeman** folder, maybe in text format, as a convenient backup option.
I know they can be exported in .json format, but that isn't easy to work with, for instance grepping inside the file or select/copy doesn't work very well for me.

As a related issue, I think it would also be nice to be able to highlight/copy individual rules on the *about:policeman#user-rulesets* page: that may come handy if you eventually want to copy some blocked URLs and then add them to a hostfile or some adblocker filter list.

If it would be better to open a separate issue for this request, let me know.

**Comment by dxdragon** [comment]
@futpib What's the progress on this project? any plan to release a new version before new year? at least to fix the detects or bugs...

**Comment by heforfree** [comment]
guys go and fill survey"survey regarding it's future"
If you like Policeman, please consider participating in a survey regarding it's future.

https://outgoing.prod.mozaws.net/v1/acdee26bcc62f0b5616c9c4121f6fb9e2765f62a8ac1e6343a48b8bde91f844d/https%3A//goo.gl/forms/g6ldXl18sUz3eIwW2
according here https://addons.mozilla.org/en-US/firefox/addon/policeman/

**Comment by muellert** [comment]
How can I import the rules created by Requestpolicy's Export feature? Because importing only from an installed Requestpolicy addon is imho not useful - I had Requestpolicy 0.5x, 1.x, and Policeman installed, then de-installed Requestpolicy because there was a warning about duplicate instances, which seems to have been caused by policeman, then wanted to import from Requestpolicy, but only got their standard handful of "Allow" rules to import. NOT what I wanted - I have ~1800 rules, actually.

**Comment by myrdd** [comment]
> I had Requestpolicy 0.5x, 1.x, and Policeman installed, then de-installed Requestpolicy because there was a warning about duplicate instances

You saw the warning because it's not possible to use 0.5 and 1.x at the same time.

**Comment by muellert** [comment]
Well, I knew that and had 0.5 disabled already. The message suggested that disabling it should be enough, so I concluded that the conflict message should have been caused by having both Requestpolicy and Policeman activated at the same time.

**Comment by futpib** [comment]
Brings back memories, thanks to everyone involved! :heart:
